hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/AbstractEmbeddableInitializer.java

```diff
@@ -422,18 +440,24 @@ private Object determineParentInstance(RowProcessingState processingState) {
 			return null;
```

Comment:
Can you try to replacing this if-block with `assert parentPath != null;`? It seems impossible to me to have a `null` parent navigable path.

Reply:
done